open-metadata-implementation/server-chassis/server-chassis-spring/build.gradle

```diff
@@ -44,59 +44,66 @@ dependencies {
     runtimeOnly 'org.hibernate:hibernate-validator'
     runtimeOnly project(':open-metadata-implementation:platform-services:platform-services-spring')
     runtimeOnly project(':open-metadata-implementation:admin-services:admin-services-spring')
-    runtimeOnly project(':open-metadata-implementation:framework-services:ocf-metadata-management:ocf-metadata-spring')
-    runtimeOnly project(':open-metadata-implementation:repository-services:repository-services-spring')
```

Comment:
The truely optional services are the registered services - these are the OMASs (access-services), OMVSs (view-services), OMIS (integration-services) and OMES (engine-services).  Their are loaded into the platform and initialized when their spring module is detected with the component scan.  Excluding their spring module from the loader path ensures none of the associated modules from the service are loaded.  Trying to activate them through a configuration document results in an error.  This ability to completely remove registered services is what we have documented at the moment.

This build option is also the way that new, third-party registered services are introduced.  Their modules are added to the loader path - detected in component scan and loaded as if part of the platform build.  They can alsothen  be configured in a server just like the Egeria provided registered services.

The new (fixed) services listed - the framework services, conformance test services and governance server services - operate differently.  Their server/api modules are always needed by the platform.  If we make their spring modules "compileOnly" and they are not in the loader path, then their REST calls are not available.  Their services can be initialized and started in the platform through configuration documents.  However, the REST API is not available.  This does have some value in that it reduces the attack surface of the platform.

So two choices:
* remove the new fixed services from the list in this PR
* update the documentation on this profile option to explain why/when/how to remove the REST API support for the fixed services.

I favour the second option and volunteer to do the doc updates.  Do you agree?

Reply:
Yes. Can we also check the other appropriate jars are present. This is the current list of jars from egeria that is found in the cut-down chassis, as produced by gradle:
```
admin-services-api-3.15-SNAPSHOT.jar
admin-services-client-3.15-SNAPSHOT.jar
admin-services-registration-3.15-SNAPSHOT.jar
admin-services-server-3.15-SNAPSHOT.jar
admin-services-spring-3.15-SNAPSHOT.jar
asset-lineage-api-3.15-SNAPSHOT.jar
asset-lineage-client-3.15-SNAPSHOT.jar
asset-lineage-topic-connectors-3.15-SNAPSHOT.jar
audit-log-framework-3.15-SNAPSHOT.jar
basic-file-connector-3.15-SNAPSHOT.jar
connector-configuration-factory-3.15-SNAPSHOT.jar
csv-file-connector-3.15-SNAPSHOT.jar
data-engine-api-3.15-SNAPSHOT.jar
data-engine-client-3.15-SNAPSHOT.jar
data-engine-proxy-connector-3.15-SNAPSHOT.jar
data-engine-proxy-services-server-3.15-SNAPSHOT.jar
data-engine-topic-connectors-3.15-SNAPSHOT.jar
engine-host-services-api-3.15-SNAPSHOT.jar
engine-host-services-registration-3.15-SNAPSHOT.jar
engine-host-services-server-3.15-SNAPSHOT.jar
ffdc-services-3.15-SNAPSHOT.jar
gaf-metadata-api-3.15-SNAPSHOT.jar
gaf-metadata-client-3.15-SNAPSHOT.jar
gaf-metadata-server-3.15-SNAPSHOT.jar
generic-handlers-3.15-SNAPSHOT.jar
governance-action-framework-3.15-SNAPSHOT.jar
governance-engine-api-3.15-SNAPSHOT.jar
governance-engine-client-3.15-SNAPSHOT.jar
governance-engine-topic-connectors-3.15-SNAPSHOT.jar
http-helper-3.15-SNAPSHOT.jar
integration-daemon-services-api-3.15-SNAPSHOT.jar
integration-daemon-services-registration-3.15-SNAPSHOT.jar
integration-daemon-services-server-3.15-SNAPSHOT.jar
metadata-security-apis-3.15-SNAPSHOT.jar
metadata-security-connectors-3.15-SNAPSHOT.jar
metadata-security-server-3.15-SNAPSHOT.jar
multi-tenant-3.15-SNAPSHOT.jar
ocf-metadata-api-3.15-SNAPSHOT.jar
ocf-metadata-client-3.15-SNAPSHOT.jar
ocf-metadata-server-3.15-SNAPSHOT.jar
open-connector-framework-3.15-SNAPSHOT.jar
open-discovery-framework-3.15-SNAPSHOT.jar
open-lineage-services-api-3.15-SNAPSHOT.jar
open-lineage-services-server-3.15-SNAPSHOT.jar
open-metadata-archive-file-connector-3.15-SNAPSHOT.jar
open-metadata-conformance-suite-api-3.15-SNAPSHOT.jar
open-metadata-conformance-suite-server-3.15-SNAPSHOT.jar
open-metadata-types-3.15-SNAPSHOT.jar
platform-services-api-3.15-SNAPSHOT.jar
platform-services-server-3.15-SNAPSHOT.jar
platform-services-spring-3.15-SNAPSHOT.jar
repository-handler-3.15-SNAPSHOT.jar
repository-services-apis-3.15-SNAPSHOT.jar
repository-services-archive-utilities-3.15-SNAPSHOT.jar
repository-services-implementation-3.15-SNAPSHOT.jar
rest-client-connectors-api-3.15-SNAPSHOT.jar
rest-client-factory-3.15-SNAPSHOT.jar
spring-rest-client-connector-3.15-SNAPSHOT.jar
```

Reply:
has been checked. closing conversation.